packages/core/src/credential/Credential.ts

```diff
@@ -232,6 +233,35 @@ export async function verifySignature(
     throw new SDKErrors.SignatureUnverifiableError(
       'Challenge differs from expected'
     )
+
+  // check if credential owner matches signer
+  const { owner } = input.claim
+  const {
+    did: signer,
+    address: signerAddress,
+    type: signerType,
+    authKeyTypeEncoding: signerKeyType,
+  } = parse(input.claimerSignature.keyUri)
+  if (owner !== signer) {
+    const {
+      address: ownerAddress,
+      type: ownerType,
+      authKeyTypeEncoding: ownerKeyType,
+    } = parse(owner)
+
+    if (ownerType === 'full') {
+      // owner did is full did -> exact match required (signer cannot be light did)
+      throw new SDKErrors.DidSubjectMismatchError(signer, owner)
+    } else {
+      // owner did is light did -> full- or light did with same subject may sign
+      if (ownerAddress !== signerAddress)
+        throw new SDKErrors.DidSubjectMismatchError(signer, owner)
+      // if signer is also a light did, require both to use the same key type
+      if (signerType === 'light' && ownerKeyType !== signerKeyType)
+        throw new SDKErrors.DidSubjectMismatchError(signer, owner)
+    }
+  }
+
```

Comment:
The logic is fine, however do you think something like that will be more elegant? 
```suggestion

  const owner = parse(input.claim.owner)
  const signer = parse(input.claimerSignature.keyUri)

  // check if credential owner matches signer
  if (owner.did !== signer.did) {
    // owner did is full did -> exact match required (signer cannot be light did)
    const typesMismatch = owner.type === 'full'

    // owner did is light did -> full- or light did with same subject may sign
    const subjectMismatch = owner.address !== signer.address

    // if signer is also a light did, require both to use the same key type
    const keyTypeMismatch =
      signer.type === 'light' &&
      owner.authKeyTypeEncoding !== signer.authKeyTypeEncoding

    if (typesMismatch || subjectMismatch || keyTypeMismatch) {
      throw new SDKErrors.DidSubjectMismatchError(signer.did, owner.did)
    }
  }
```

Reply:
oh nice, that does look better indeed